src/sinks/loki/sink.rs

```diff
@@ -776,14 +919,85 @@ mod tests {
         assert!(!String::from_utf8_lossy(&record.event.event).contains("value"));
     }
 
+    #[test]
+    fn encoder_with_structured_metadata() -> Result<(), serde_json::Error> {
+        let mut structured_metadata = HashMap::default();
+        structured_metadata.insert(
+            Template::try_from("pod_labels_*").unwrap(),
+            Template::try_from("{{ kubernetes.pod_labels }}").unwrap(),
+        );
+        structured_metadata.insert(
+            Template::try_from("*").unwrap(),
+            Template::try_from("{{ metadata }}").unwrap(),
+        );
+        structured_metadata.insert(
+            Template::try_from("cluster_name").unwrap(),
+            Template::try_from("static_cluster_name").unwrap(),
+        );
+
+        let mut encoder = EventEncoder {
+            key_partitioner: KeyPartitioner::new(None),
+            transformer: Default::default(),
+            encoder: Encoder::<()>::new(JsonSerializerConfig::default().build().into()),
+            labels: HashMap::default(),
+            structured_metadata: Option::None,
```

Comment:
Seems like you should actually be setting this to the `structured_metadata` field you've constructed above?